lib/presentation/profile/profile_screen.dart

```diff
@@ -129,21 +129,59 @@ class _UserProfilePageState extends State<UserProfilePage> {
                                     maxRadius: 50,
                                   ),
                                 ),
-                                if (state.isEditing == true) ...[
+                                if (state.userProfile != null) ...[
                                   Positioned(
                                     bottom: 0,
                                     right: 0,
                                     child: FloatingActionButton(
-                                      onPressed: () =>
-                                          PhotoSelector.showPhotoSelector(
-                                        context,
-                                        onSelected: (image) {
-                                          setState(() => _image = image);
-                                        },
-                                      ),
+                                      onPressed: () {
+                                        PhotoSelector.showPhotoSelector(
+                                          context,
+                                          onSelected: (image) {
+                                            setState(() => _image = image);
+                                          },
+                                        );
+                                        context
+                                            .read<ProfileBloc>()
+                                            .add(EditProfilePic());
```

Comment:
I don't remember how this works, but shouldn't the image be passed to this event?